channel/src/fs_verifier_channel.rs

```diff
@@ -148,24 +146,26 @@ impl<F: PrimeField, P: Prng, W: Digest> VerifierChannel for FSVerifierChannel<F,
         Ok(bytes)
     }
 
-    fn recv_commit_hash(&mut self) -> Result<Self::Commitment, anyhow::Error> {
-        let size = std::mem::size_of::<Self::Commitment>();
-        let bytes = self.recv_bytes(size)?;
+    fn recv_commit_hash_default(&mut self) -> Result<Vec<u8>, anyhow::Error> {
+        self.recv_commit_hash(P::digest_size())
+    }
 
-        let commitment = GenericArray::from_exact_iter(bytes).unwrap();
+    fn recv_commit_hash(&mut self, size: usize) -> Result<Vec<u8>, anyhow::Error> {
+        let bytes = self.recv_bytes(size)?;
 
         self.states.increment_commitment_count();
         self.states.increment_hash_count();
-        Ok(commitment)
+        Ok(bytes)
     }
 
-    fn recv_decommit_node(&mut self) -> Result<Self::Commitment, anyhow::Error> {
-        let size = std::mem::size_of::<Self::Commitment>();
-        let bytes = self.recv_bytes(size)?;
+    fn recv_decommit_node_default(&mut self) -> Result<Vec<u8>, anyhow::Error> {
```

Comment:
The output type changes from `Self::Commitment` to an unmarked `Vec<u8>`. 
To improve readability, consider wrapping the vector in an enum or struct.